nemo_curator/filters/code.py

```diff
@@ -102,7 +102,10 @@ def keep_document(self, score):
 class TokenizerFertilityFilter(DocumentFilter):
 
     def __init__(self, path_to_tokenizer=None, min_char_to_token_ratio=2.5):
-        from nemo.collections.common.tokenizers import SentencePieceTokenizer
+        try:
+            from nemo.collections.common.tokenizers import SentencePieceTokenizer
+        except (ImportError, ModuleNotFoundError):
+            from .sentencepiece_tokenizer import SentencePieceTokenizer
```

Comment:
What do we think about this?

Reply:
Based on our discussions from slack, I think we can just transform this class to be something like this:
```python
class TokenizerFertilityFilter(DocumentFilter):

    def __init__(self, path_to_tokenizer=None, min_char_to_token_ratio=2.5):
        if path_to_tokenizer is None:
            raise ValueError(
                "Must provide a valid path to a SentencePiece " "tokenizer"
            )
        self._tokenizer = sentencepiece.SentencePieceProcessor()
        self._tokenizer.Load(path_to_tokenizer)
        self._threshold = min_char_to_token_ratio

        self._name = "tokenizer_fertility"

    def score_document(self, source):
        tokens = self._tokenizer.encode_as_pieces(source)
        num_chars = len(source)
        num_tokens = len(tokens)
        if num_tokens == 0:
            return -1
        return num_chars / num_tokens

    def keep_document(self, score):
        return score >= self._threshold
```

Then we can just delete the one file you copied over. Lmk what you think.

Reply:
We should probably run this via a batch instead of running it on a per file pieces and return a single file. We can also probably use `crossfit` for it (if we want to)

Reply:
This is what that will look like. 
```
 cf.op.Tokenizer(model, cols=["text"], tokenizer_type="sentencepiece")
 ```
 
 That said, we might have to ensure this works in a CPU environment too so there might be some complexity here we need to fix.

Reply:
Thanks @VibhuJawa ! I have opened https://github.com/NVIDIA/NeMo-Curator/issues/377 to track this.